app/main.py

```diff
@@ -1,5 +1,81 @@
 class Cargo:
-    def __init__(self, weight: int) -> None:
+    def __init__(
+            self,
+            weight: int
+    ) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: None | list = None
+    ) -> None:
+
+        if coords is None:
+            coords = [0, 0]
+
+        self.name = name
+        self.weight = weight
+        self.coords = coords
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight:
+            int,
+            coords: None | list = None
+    ) -> None:
+
+        if coords is None:
+            coords = [0, 0, 0]
+
+        super().__init__(name=name, weight=weight, coords=coords)
```

Comment:
Perform the assignment directly in the `super().__init__(`
```suggestion
        super().__init__(name=name, weight=weight, coords=coords or [0, 0, 0])
```